cpp/src/parquet/metadata.cc

```diff
@@ -651,9 +651,12 @@ class FileMetaData::FileMetaDataImpl {
     std::string key = file_decryptor_->GetFooterKey();
     std::string aad = encryption::CreateFooterAad(file_decryptor_->file_aad());
 
-    auto aes_encryptor = encryption::AesEncryptor::Make(
-        file_decryptor_->algorithm(), static_cast<int>(key.size()), true,
-        false /*write_length*/, nullptr);
+    auto aes_encryptor = encryption::AesEncryptor::Make(file_decryptor_->algorithm(),
+                                                        static_cast<int>(key.size()),
+                                                        true, false /*write_length*/);
+    if (ARROW_PREDICT_FALSE(aes_encryptor == nullptr)) {
```

Comment:
I think we can remove this check now that we know `AesEncryptor::Make` will never return a null pointer?